tower-buffer/src/lib.rs

```diff
@@ -437,16 +509,23 @@ where
                             }
                             // We may want to also make progress on current requests
                         }
-                        Err(_) => {
-                            self.state.open.store(false, Ordering::Release);
-                            return Ok(().into());
```

Comment:
I think no longer finishing the worker future in this case seems correct, but it does look like a possible change in behaviour here --- the worker will remain alive until the service is dropped...

Reply:
Ah, no, the worker will still exit. It just won't exit until it has sent errors out to all outstanding requests. Specifically, `self.failed` will close `self.rx`, which means that `poll_next_msg` will eventually return `Ready(None)`, which will set `self.finish = true`, which will cause the `Worker` to exit. We need to not exit immediately (like the old code did), because then we'd just be dropping the `oneshot::Sender` for any outstanding requests, which would then not get notified of the error!

Reply:
Thanks for clearing that up, the flow got a little more complicated here. I thought that the worker would continue to error any _new_ requests, but I think I misread it.

Reply:
Yeah, new requests will see the error by observing that the request `mpsc` has been closed, and then reading out the `Arc<Error>` that the `Worker` sets before closing the receive end of the channel